src/Microsoft.ML.Core/Data/ModelHeader.cs

```diff
@@ -351,10 +351,17 @@ public static void CheckVersionInfo(ref ModelHeader header, VersionInfo ver)
             Contracts.CheckDecode(header.ModelSignature == ver.ModelSignature, "Unknown file type");
             Contracts.CheckDecode(header.ModelVerReadable <= header.ModelVerWritten, "Corrupt file header");
             if (header.ModelVerReadable > ver.VerWrittenCur)
-                throw Contracts.ExceptDecode("Cause: ML.NET {0} cannont read component '{1}' of the model, because the model is too new.\n" +
+                throw Contracts.ExceptDecode("Cause: ML.NET {0} cannot read component '{1}' of the model, because the model is too new.\n" +
                                 "Suggestion: Make sure the model is trained with ML.NET {0} or older.\n" +
                                 "Debug details: Maximum expected version {2}, got {3}.",
                                 typeof(VersionInfo).Assembly.GetName().Version, ver.LoaderSignature, header.ModelVerReadable, ver.VerWrittenCur);
+            if (header.ModelVerWritten > ver.VerWrittenCur)
+            {
+                throw Contracts.ExceptDecode("The model was trained on a newer version of ML․NET.\n" +
```

Comment:
What do you think about putting this exception behind a flag that the user can shut it off? That way if someone is relying on this functionality, and it happens to work for them, they can turn the exception off and get unblocked. Potentially just an environment variable or something would be enough.

Reply:
sounds good! Let me add it.

Reply:
Existing environment variables start with `MICROSOFTML_`. Maybe we should follow the same naming pattern?

https://github.com/dotnet/machinelearning/blob/d682a602d7db3169b13693d3064560bf5341cf61/src/Microsoft.ML.Core/Utilities/ResourceManagerUtils.cs#L50-L59

https://github.com/dotnet/machinelearning/blob/b861b5d64841cbe0f2c866ee7586872aac450a51/src/Microsoft.ML.Core/Utilities/PathUtils.cs#L14-L17

Reply:
Thanks for pointing that out. Would "MICROSOFT_NEWER_MODEL_VERSION_ALLOWED" be a good name?